src/identity.js

```diff
@@ -721,6 +730,45 @@ export class Identity extends EventEmitter {
         return null;
     }
 
+
+    /**
+     * Starts sms/otp/password login flow for logged in users. Flow results with user access
+     * token with intended methods in amr field (make sure to verify token amr).
+     * @summary Triggers sms/otp/password authentication
+     * @see https://tools.ietf.org/html/rfc6749#section-4.1.1
+     * @param {object} options
+     * @param {string} [options.acrValues] - Authentication Context Class Reference Values. Possible
+     * values are `sms`, `otp`, `password`. Might be combined and separate by space.
+     * @param {string} options.state - An opaque value used by the client to maintain state between
+     * the request and callback. It's also recommended to prevent CSRF
+     * @see https://tools.ietf.org/html/rfc6749#section-10.12
+     * @param {string} [options.scope='openid'] - The OAuth scopes for the tokens. This is a list of
+     * scopes, separated by space. If the list of scopes contains `openid`, the generated tokens
+     * includes the id token which can be useful for getting information about the user. Omitting
+     * scope is allowed, while `invalid_scope` is returned when the client asks for a scope you
+     * aren’t allowed to request.
+     * @see https://tools.ietf.org/html/rfc6749#section-3.3
+     * @param {string} [options.redirectUri=this.redirectUri] - Redirect uri that will receive the
+     * code. Must exactly match a redirectUri from your client in self-service
+     * @param {string} [options.tag=''] - Pulse tag
+     * @param {string} [options.locale=''] - Optional parameter to overwrite client locale setting.
+     * New flows supports nb_NO, fi_FI, sv_SE, en_US
+     */
+    assertAuthorizationMethod({
```

Comment:
Very good question. That flow will be run with different layout than normal flow. It suppose to be run only for logged in users, to ensure additional assertion- not 2fa since in most cases it will be just sms/otp. It also doesn't support many options that login does.
Since it suppose to solve different business logic I think it would be better to have it as separate function. WDYT?

Reply:
It's intended to be used for already logged-in user, but there's no internal check (e.g. call to `hasSession`) in this function. This means the user will be redirected to full MFA login if they're not already logged-in, right? Then this function could be used for the same thing as `login()` and there's no difference between them.

Reply:
Also, if keeping as separate function the naming is a bit confusing:
This function doesn't "assert" anything (client's need to do the final check in ID token) and it should be "authentication" instead of "authorization". Maybe simply naming it `reauthenticate()` or similar, WDYT?